File: 02_10_2024_Rank_Transform_of_an_Array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int> ans;
        map<int,int> mpp;
        for(int i:arr)
        {
            mpp[i]++;
        }
        int m = 1;
        for(auto it:mpp)
        {
            mpp[it.first] = m++;
        }
        for(int i :arr)
        {
            ans.push_back(mpp[i]);
        }
        return ans;
    }
};

File: 02_10_2024_Rank_Transform_of_an_Array.java
import java.util.*;

class Solution {
    public int[] arrayRankTransform(int[] arr) {
        List<Integer> ans = new ArrayList<>();
        Map<Integer, Integer> mpp = new HashMap<>();
        
        for (int i : arr) {
            mpp.put(i, mpp.getOrDefault(i, 0) + 1);
        }
        
        int m = 1;
        for (Map.Entry<Integer, Integer> it : mpp.entrySet()) {
            mpp.put(it.getKey(), m++);
        }
        
        for (int i : arr) {
            ans.add(mpp.get(i));
        }
        
        return ans.stream().mapToInt(i -> i).toArray();
    }
}

